pkg/datapath/linux/probes/probes.go

```diff
@@ -487,3 +465,149 @@ func HaveV3ISA() error {
 	}
 	return nil
 }
+
+// CreateHeaderFiles creates C header files with macros indicating which BPF
+// features are available in the kernel.
+func CreateHeaderFiles(headerDir string, probes *FeatureProbes) error {
+	common, err := os.Create(filepath.Join(headerDir, "features.h"))
+	if err != nil {
+		return fmt.Errorf("could not create common features headerfile: %w", err)
+	}
+	defer common.Close()
+	if err := writeCommonHeader(common, probes); err != nil {
+		return fmt.Errorf("could not write common features headerfile: %w", err)
+	}
+
+	skb, err := os.Create(filepath.Join(headerDir, "features_skb.h"))
+	if err != nil {
+		return fmt.Errorf("could not create skb related features headerfile: %w", err)
+	}
+	defer skb.Close()
+	if err := writeSkbHeader(skb, probes); err != nil {
+		return fmt.Errorf("could not write skb related features headerfile: %w", err)
+	}
+
+	xdp, err := os.Create(filepath.Join(headerDir, "features_xdp.h"))
+	if err != nil {
+		return fmt.Errorf("could not create xdp related features headerfile: %w", err)
+	}
+	defer xdp.Close()
+	if err := writeXdpHeader(xdp, probes); err != nil {
+		return fmt.Errorf("could not write xdp related features headerfile: %w", err)
+	}
+
+	return nil
+}
+
+// ExecuteHeaderProbes probes the kernel for a specific set of BPF features
+// which are currently used to generate various feature MACROS for the datapath.
+// The probe results returned in FeatureProbes are then used in the respective
+// function that writes the actual C MACRO definitions.
+// Further needed probes should be added here.
+func ExecuteHeaderProbes() *FeatureProbes {
```

Comment:
This can be called from the privileged test suite for good measure. `HaveXYZ()` already panics, so just assert it doesn't return nil.